### PR TITLE
Task-1: Adding static trocadilhos list

### PR DESCRIPTION
**Pull request by gabrielcicero45**
- Adds the `TrocadilhosList` and `Trocadilhos`  component.
image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,9 +1,33 @@
...
+  votes: 10,
+  dev: "cicero.medeirios",
+  context: "Lorem ipsum dolor,sit alit. Fugiat animi quaerat assumenda architecto possimus qui itaque reprehenderit tempore provident debitis? Dicta blanditiis mollitia reiciendis saepe. Alias, facere? Excepturi, officia obcaecati.",
+  message: "Lorem ipsus"
+  },{date: "13/02/2023",
+  votes: 10,
+  dev: "cicero.medeirios",
+  context: "Lorem ipsum dolor,sit alit. Fugiat animi quaerat assumenda architecto possimus qui itaque reprehenderit tempore provident debitis? Dicta blanditiis mollitia reiciendis saepe. Alias, facere? Excepturi, officia obcaecati.",
+  message: "Lorem ipsus"
+  },]
```
Comment (tonyaraujop) [inline src/App.js]: Please think of this as an entry data, that can be fetch from an API endpoint. Move this to another place to keep this file clean and easy read.

File: src/components/Trocadilho/Trocadilho.test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { render, screen } from "@testing-library/react";
+import Trocadilho from "./Trocadilho";
+
+describe("<Trocadilho/>", () => {
+  it.todo("Trocadilho must contain the date that it was made, in mm/dd/yyyy format");
+  it.todo("Trocadilho must contain the number of votes");
+  it.todo("Trocadilho must contain the Discord user who made the trocadilho");
+  it.todo("Trocadilho must contain the message");
+  it.todo("Trocadilho must contain the context");
+  it.todo('Clicking the vote button should update the vote count.')
```
Comment (tonyaraujop) [inline src/components/Trocadilho/Trocadilho.test.js]: Add the tests for this compoment

File: src/components/TrocadilhoList/TrocadilhoList.test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      context:
+        "Lorem ipsum dolor,sit alit. Fugiat animi quaerat assumenda architecto possimus qui itaque reprehenderit tempore provident debitis? Dicta blanditiis mollitia reiciendis saepe. Alias, facere? Excepturi, officia obcaecati.",
+      message: "Lorem ipsus",
+    },
+  ];
+  const trocadilhosEmpty = [];
+  it("A list of available trocadilhos should be displayed on the page", () => {
+    render(<TrocadilhoList trocadilhos={trocadilhosAvailable} />);
+
+    screen.logTestingPlaygroundURL();
```
Comment (tonyaraujop) [inline src/components/TrocadilhoList/TrocadilhoList.test.js]: Do you need this for testing?

File: src/components/TrocadilhoList/TrocadilhoList.test.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    const trocadilhosTable = screen.getByRole("table");
+
+    expect(trocadilhosTable).toBeInTheDocument();
```
Comment (tonyaraujop) [inline src/components/TrocadilhoList/TrocadilhoList.test.js]: You should also have a expect for the table size (how many trocadilhos you have in one list).

**Review by tonyaraujop — CHANGES_REQUESTED** [review]
Some minor request changes

File: src/components/Pun/Pun.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  const [votes, setVotes] = useState(initialVotes)
+  return (
+    <tr>
+      <td>{date}</td>
+      <td>{votes}</td>
+      <td>{dev}</td>
+      <td>{context}</td>
+      <td>{message}</td>
+      <td>
+        <button onClick={()=>{setVotes(votes => votes+=1)}}>Votar</button>
```
Comment (tonyaraujop) [inline src/components/Pun/Pun.js:14]: You can make a function and use it here. It's going to be easier to test the behavior and make sure the function is being called when the button is clicked. You can have a `utils` to store this function and use it here.

File: src/components/Pun/Pun.test.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        dev={pun.dev}
+        context={pun.context}
+        message={pun.message}
+      />
+    );
+    const punVotes = screen.getByRole("cell", { name: 10 });
+
+    expect(punVotes).toBeInTheDocument();
+  });
+  it("Contains the Discord user who made the Pun", () => {
```
Comment (tonyaraujop) [inline src/components/Pun/Pun.test.js]:
```suggestion
  });
  
  it("Contains the Discord user who made the Pun", () => {
```

File: src/components/Pun/Pun.test.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    render(
+      <Pun
+        date={pun.date}
+        initialVotes={pun.votes}
+        dev={pun.dev}
+        context={pun.context}
+        message={pun.message}
+      />
+    );
+    const punVotes = screen.getByRole("cell", { name: 10 });
```
Comment (tonyaraujop) [inline src/components/Pun/Pun.test.js:40]:
```suggestion
    );
    
    const punVotes = screen.getByRole("cell", { name: 10 });
```

File: src/components/Pun/Pun.test.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    render(
+      <Pun
+        date={pun.date}
+        initialVotes={pun.votes}
+        dev={pun.dev}
+        context={pun.context}
+        message={pun.message}
+      />
+    );
+    const punDate = screen.getByRole("cell", { name: "01/02/2023" });
```
Comment (tonyaraujop) [inline src/components/Pun/Pun.test.js:25]:
```suggestion
    );
    
    const punDate = screen.getByRole("cell", { name: "01/02/2023" });
```

File: src/components/Pun/Pun.test.js [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    render(
+      <Pun
+        date={pun.date}
+        initialVotes={pun.votes}
+        dev={pun.dev}
+        context={pun.context}
+        message={pun.message}
+      />
+    );
+    const punDev = screen.getByRole("cell", { name: /cicero\.medeirios/i });
```
Comment (tonyaraujop) [inline src/components/Pun/Pun.test.js:55]:
```suggestion
    );
    
    const punDev = screen.getByRole("cell", { name: /cicero\.medeirios/i });
```

File: src/components/Pun/Pun.test.js (hunk 6 above)
Comment (tonyaraujop) [inline src/components/Pun/Pun.test.js]: The propertie is `dev` why the test is for discord user?

File: src/components/Pun/Pun.test.js [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+      />
+    );
+    const punContext = screen.getByRole("cell", {
+      name: "Lorem ipsum dolor,sit alit. Fugiat animi quaerat assumenda architecto possimus qui itaque reprehenderit tempore provident debitis? Dicta blanditiis mollitia reiciendis saepe. Alias, facere? Excepturi, officia obcaecati.",
+    });
+
+    expect(punContext).toBeInTheDocument();
+  });
+
+  it("Clicking the vote button should update the vote count.", ()=>{
```
Comment (tonyaraujop) [inline src/components/Pun/Pun.test.js]: You miss a test to render the button.

**Review by tonyaraujop — CHANGES_REQUESTED** [review]
Another few tips to adjust this PR. Great work so far.

File: src/components/Pun/Pun.js (hunk 5 above)
Comment (bioneoficial) [reply src/components/Pun/Pun.js:14]: If you have time, its nice create an component only for this button.
VoteButton Folder with VoteButton.js and VoteButton.test.js for example

File: src/components/PunsList/PunsList.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+                  date={date}
+                  initialVotes={votes}
+                  dev={dev}
+                  context={context}
+                  message={message}
+                />
+              );
+            })
+          ) : (
+            <h1>Lista vazia!</h1>
```
Comment (bioneoficial) [inline src/components/PunsList/PunsList.js]: Não é legal termos mais de H1 na página, seria interessante usar um H2/H3/H4 ou P, pela semantica.

**Review by bioneoficial — APPROVED** [review]

**Review by bioneoficial — APPROVED** [review]

**Review by julialbq — APPROVED** [review]
I learned a lot from your tests!

**Review by tonyaraujop — APPROVED** [review]
Great work.

**Review by bioneoficial — COMMENTED** [review]
Thats a nice PR!!

**Pull request merged**